lib/logic/player/content.dart

```diff
@@ -383,73 +310,75 @@ class ContentControl extends Control {
   //   ]);
   // }
 
-  /// Returns content of specified type.
-  List<T> getContent<T extends Content>({
-    Type? contentType,
+  /// Returns content of specified [contentType].
+  List<T> getContent<T extends Content>(
+    ContentType contentType, {
     bool filterFavorite = false,
   }) {
+    final List<T> contentList;
+    switch (contentType) {
+      case ContentType.song:
+        contentList = state.allSongs.songs as List<T>;
+        break;
+      case ContentType.album:
+        contentList = state.albums.values.toList() as List<T>;
+        break;
+      case ContentType.playlist:
+        contentList = state.playlists as List<T>;
+        break;
+      case ContentType.artist:
+        contentList = state.artists as List<T>;
+        break;
+    }
     if (filterFavorite) {
-      return contentPick<T, ValueGetter<List<T>>>(
-        contentType: contentType,
-        song: () => ContentUtils.filterFavorite(state.allSongs.songs).toList() as List<T>,
-        album: () => ContentUtils.filterFavorite(state.albums.values).toList() as List<T>,
-        playlist: () => ContentUtils.filterFavorite(state.playlists).toList() as List<T>,
-        artist: () => ContentUtils.filterFavorite(state.artists).toList() as List<T>,
-      )();
+      return ContentUtils.filterFavorite(contentList).toList();
     }
-    return contentPick<T, ValueGetter<List<T>>>(
-      contentType: contentType,
-      song: () => state.allSongs.songs as List<T>,
-      album: () => state.albums.values.toList() as List<T>,
-      playlist: () => state.playlists as List<T>,
-      artist: () => state.artists as List<T>,
-    )();
+    return contentList;
   }
 
   /// Returns content of specified type with ID.
-  T? getContentById<T extends Content>(int id, [Type? contentType]) {
-    if ((contentType ?? T) == Album) {
+  T? getContentById<T extends Content>(int id, ContentType<T> contentType) {
```

Comment:
Oh, so it's able to imply the generics here. And it's also a compile time error to have something like `getContentById<Song>(entry.id, ContentType.album)`. Very cool.